### PR TITLE
Created recipe for SHARPpy v1.4.0 beta 1 - first Conda-forge submission

### PR DESCRIPTION
**Pull request by wblumberg**
Checklist
- [X] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml" 
- [X] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [X] Source is from official source
- [X] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [X] If static libraries are linked in, the license of the static library is packaged.
- [X] Build number is 0
- [X] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [X] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

Hi there!  This is a pull request to submit a beta version of the SHARPpy package (used widely by the atmospheric science community) to the conda-forge channel.  I haven't submitted to conda-forge before, but from what I'm reading and hearing it sounds like a much easier process to deploy this package to other users!  Please let me know if there are things I can do to improve and make it easier for the wonderful people who manage this channel!

### CODE REVIEW

File: recipes/sharppy/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    - python-dateutil
+    - certifi
+    
+test:
+  imports:
+    - sharppy
+
+about:
+  home: https://github.com/sharppy/SHARPpy
+  license: BSD
```
Comment (scopatz) [inline recipes/sharppy/meta.yaml]: Which version of BSD? Also this needs a license_family entry

Comment (wblumberg) [reply recipes/sharppy/meta.yaml]: I'll take care of that now!

File: recipes/sharppy/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  script: "{{ PYTHON }} -m pip install . --no-deps -vv --ignore-installed --no-cache-dir"
+  osx_is_app: True
+  entry_points:
+    - sharppy = runsharp.full_gui:main
+
+requirements:
+  host:
+    - python >=3.6
+    - setuptools
+    - pip
```
Comment (scopatz) [inline recipes/sharppy/meta.yaml]: I think you don't need setuptools, if you have pip

Comment (wblumberg) [reply recipes/sharppy/meta.yaml]: Just removed it.  I should be pushing a commit with these changes in a second.

**Comment by wblumberg** [comment]
@scopatz - I'm not sure why CI on Appveyor keeps failing.  It's doing something with Python 2.7, which this package doesn't use.

**Comment by scopatz** [comment]
You don't need to worry about appveyor if azure on windows passes

**Comment by wblumberg** [comment]
@scopatz - I got everything passing except for Appveyor now.

**Comment by tsupinie** [comment]
I'd like to be a maintainer for this recipe. @wblumberg set me.

File: .azure-pipelines/azure-pipelines-linux.yml [hunk 3]
```diff
@@ -14,7 +14,12 @@ jobs:
   steps:
   - script: |
       # sudo pip install --upgrade pip
+      #sudo apt-get update
+      #sudo yum install mesa-libGL.x86_64
+      #sudo apt-get install -y libgl1-mesa-dev
+      #sudo apt-get install qtbase5-dev
       sudo pip install setuptools shyaml
+      #export DISPLAY=:0.0
```
Comment (scopatz) [inline .azure-pipelines/azure-pipelines-linux.yml]: Please do not modify files outside of the recipes directory

File: recipes/sharppy/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+        It was developed to provide the atmospheric science community a free and consistent source of sounding analysis 
+        routines. SHARPpy is updated and vetted by professional meteorologists and climatologists within the scientific
+        community to help maintain a standard source of sounding routines.
+  doc_url: https://sharppy.github.io/SHARPpy/index.html 
+  dev_url: https://github.com/sharppy/SHARPpy
+
+extra:
+  #identifiers:
+  #  - doi:10.1175/BAMS-D-15-00309.1
+  #  - doi:10.1175/BAMS-D-15-00309.2
```
Comment (scopatz) [inline recipes/sharppy/meta.yaml]: please remove comments

Comment (wblumberg) [reply recipes/sharppy/meta.yaml]: On it.

File: .azure-pipelines/azure-pipelines-linux.yml (hunk 3 above)
Comment (wblumberg) [reply .azure-pipelines/azure-pipelines-linux.yml]: On it!

File: recipes/sharppy/meta.yaml (hunk 4 above)
Comment (wblumberg) [reply recipes/sharppy/meta.yaml]: Done.

File: .azure-pipelines/azure-pipelines-linux.yml (hunk 3 above)
Comment (wblumberg) [reply .azure-pipelines/azure-pipelines-linux.yml]: Done.

Comment (wblumberg) [reply .azure-pipelines/azure-pipelines-linux.yml]: My apologies.  It was left over from troubleshooting the OpenGL issue on Linux.

File: .azure-pipelines/azure-pipelines-linux.yml [hunk 5]
```diff
@@ -13,7 +13,6 @@ jobs:
   timeoutInMinutes: 360
   steps:
   - script: |
-      # sudo pip install --upgrade pip
```
Comment (scopatz) [inline .azure-pipelines/azure-pipelines-linux.yml]: Seems like you need to replace a line

Comment (wblumberg) [reply .azure-pipelines/azure-pipelines-linux.yml]: Shoot.  I'll do that.

**Comment by wblumberg** [comment]
Any updates on this pull request @scopatz ?

**Comment by scopatz** [comment]
Thanks!

**Comment by wblumberg** [comment]
Hi @scopatz ... I see that the pull request was accepted, but I haven't gotten any notifications of a feedstock generated for this package.  I'm not sure what the process is, so I looked into what was going on and it looks like the CI for the pull request was failing.  It looks like there was some kind of HTTP error when the feedstock was trying to be created - see https://travis-ci.com/conda-forge/staged-recipes/builds/151053238.

Thank you so much for reviewing our submission!

**Comment by scopatz** [comment]
CC @conda-forge/staged-recipes @conda-forge/core - I don't have bandwidth for this

**Comment by isuruf** [comment]
See https://github.com/conda-forge/sharppy-feedstock

**Pull request merged**